ibtk/include/ibtk/IndexUtilities.h

```diff
@@ -96,6 +96,29 @@ class IndexUtilities
                                                   const double* dx,
                                                   const SAMRAI::hier::Index<NDIM>& ilower,
                                                   const SAMRAI::hier::Index<NDIM>& iupper);
+	
+	/*!
+	 * \brief Convert (i,j,k,d) index for a DOF to an integer mapping on a patch level.
+	 *
+	 * \param i AMR index representing the (i,j,k) data cell index.
+	 *
+	 * \param domain_lower Lower index of the domain.
+	 *
+	 * \param num_cells Number of data array cells for a patch level.
+	 *
+	 * \param depth Data depth.
+	 *
+	 * \param offset Component (axis) offset.
+	 *
+	 * \return The linear mapping of an AMR index to a continuous non-negative 
+	 * integer space.
+	 */
+	static int getIntegerMapping(const SAMRAI::hier::Index<NDIM>& i,
```

Comment:
Index seems to suggest that something exists in a collection of things, while mapping suggests something may or may not exist. Here mapped values do not exist for any vector or operator (SAMRAI or PETSc). Hence, I preferred mapping. I can change to index if mapping makes things obscure...